spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/DefaultBinding.java

```diff
@@ -38,15 +38,22 @@
 
 	protected final T target;
 
-	protected final Lifecycle endpoint;
-
-	public DefaultBinding(String name, String group, T target, Lifecycle endpoint) {
+	protected final Lifecycle runningComponent;
+
+	/**
+	 * Creates a new instance.
+	 * @param name the name of the binding target
+	 * @param group the group (only for input targets)
+	 * @param target the binding target
+	 * @param runningComponent a {@link Lifecycle} object that is running while the binding is active
+	 *                         and will be stopped during unbinding
+	 */
+	public DefaultBinding(String name, String group, T target, Lifecycle runningComponent) {
 		Assert.notNull(target, "target must not be null");
-		Assert.notNull(endpoint, "endpoint must not be null");
 		this.name = name;
 		this.group = group;
 		this.target = target;
-		this.endpoint = endpoint;
+		this.runningComponent = runningComponent;
```

Comment:
While `runningComponent` is a good name in general here, we could as well think of some other name as it is not yet running here. May be just `lifecycleComponent` ?